File: R/elastic_net.r
library(glmnet)

DATA_PATH = "/Users/hilsts/Documents/linear-models-benchmark/data/"
df <- read.csv(paste(DATA_PATH,"song_data.csv", sep = ""))

set.seed(100)

df$song_name <- NULL

index = sample(1:nrow(df), 0.75 * nrow(df))

train = df[index, ]
test = df[-index, ]

x = as.matrix(train)
y_train = train$song_popularity

x_test = as.matrix(test)
y_test = test$song_popularity

lambdas <- 10^seq(2, -3, by = -.1)

elastic_net <- function(x, y_train, x_test) {
    en <- cv.glmnet(x, y_train, alpha = 0.5, lambda = lambdas)
    optimal_lambda <- en$lambda.min
    predictions_test <- predict(en, s = optimal_lambda, newx = x_test)
}

start_time <- Sys.time()
elastic_net(x, y_train, x_test)
end_time <- Sys.time()
print(end_time - start_time)

File: Julia/elastic_net.jl
using CSV
using DataFrames
using Lathe.preprocess: TrainTestSplit
using CSV
using DataFrames
using MLJ
using GLM
using Lasso
using MLBase




df = DataFrame(CSV.File("/Users/hilsts/Documents/linear-models-benchmark/data/song_data.csv"))
df = select!(df, Not(:song_name))
train, test = TrainTestSplit(df,.75)
X_train = convert(Matrix, train[:, Not(:song_popularity)])
y_train = train.song_popularity
X_test = convert(Matrix, test[:, Not(:song_popularity)])

function elastic_net(X_train, y_train, X_test)

    en = fit(LassoPath, X_train, y_train)
    predict(en, X_test)
end

@time elastic_net(X_train, y_train, X_test)


